### PR TITLE
Validate links

### PR DESCRIPTION
**Pull request by junrong98**
- Provide minor validation of links.
- Do not catch invalid links such as `https://.com` etc.

### CODE REVIEW

**Review by brian16600 — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/model/task/Link.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import static java.util.Objects.requireNonNull;
 
 public class Link {
+    public static final String INVALID_LINK = "The link provided is not a valid URL!";
```
Comment (junjunjieOng) [inline src/main/java/seedu/address/model/task/Link.java:6]: A suggestion, maybe can put this error message under "seedu.address.commons.core.Messages" instead. Then ins ParserUtil can import the message

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -149,11 +152,16 @@ public static Tag parseTag(String tag) throws ParseException {
     /**
      * Parses {@Code Optional<String> option} into a {@code Link}.
      */
-    public static Link parseLink(Optional<String> option) {
+    public static Link parseLink(Optional<String> option) throws ParseException {
```
[the remark this answers is not in the document]
Comment (junrong98) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:155]: No, empty link cannot be provided. It creates a Link object which contains an empty string if the users decide not to include the link in the add task command.

File: src/main/java/seedu/address/model/task/Link.java (hunk 1 above)
Comment (junrong98) [reply src/main/java/seedu/address/model/task/Link.java:6]: I'm following how these messages are implemented for the objects used in `Person` class.

Updated to `MESSAGE_CONSTRAINTS` instead.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (junrong98) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:155]: Updated accordingly, please take a look!

**Review by snss231 — COMMENTED** [review]
Nothing wrong I can see, LGTM

**Pull request merged**